### DISCUSSION TITLE
Help requested: running !whisper command yields an error in Google Colaboratory

### DISCUSSION THREAD

**Discussion by perplexed-paperclip** [openai/whisper#1222] [Q&A]
Good evening, I was hoping someone could help me decipher and resolve the error message I keep getting when I attempt to run Whisper AI using Google Colaboratory.
For context: I am not a coder. At all. The only reason I have made it this far is because I am trying to follow the instructions laid out in this YouTube video ( https://www.youtube.com/watch?v=8SQV-B83tPU ), copy-pasting all necessary code as prompted in its description. I have successfully installed Google Colaboratory and opened an .ipynb file... and to the best of my knowledge, installing Whisper was also a success. The trouble is when I attempt to run its... activation code, I believe? Pasting and running this... ` !whisper "core lecture 2.mp4" --model medium` ...has consistently returned the string below:
```
100%|██████████████████████████████████████| 1.42G/1.42G [00:10<00:00, 147MiB/s]
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/whisper/audio.py", line 46, in load_audio
    ffmpeg.input(file, threads=0)
  File "/usr/local/lib/python3.9/dist-packages/ffmpeg/_run.py", line 325, in run
    raise Error('ffmpeg', out, err)
ffmpeg._run.Error: ffmpeg error (see stderr output for detail)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/whisper", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.9/dist-packages/whisper/transcribe.py", line 448, in cli
    result = transcribe(model, audio_path, temperature=temperature, **args)
  File "/usr/local/lib/python3.9/dist-packages/whisper/transcribe.py", line 121, in transcribe
    mel = log_mel_spectrogram(audio, padding=N_SAMPLES)
  File "/usr/local/lib/python3.9/dist-packages/whisper/audio.py", line 130, in log_mel_spectrogram
    audio = load_audio(audio)
  File "/usr/local/lib/python3.9/dist-packages/whisper/audio.py", line 51, in load_audio
    raise RuntimeError(f"Failed to load audio: {e.stderr.decode()}") from e
RuntimeError: Failed to load audio: ffmpeg version 4.2.7-0ubuntu0.1 Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.4.0-1ubuntu1~20.04.1)
  configuration: --prefix=/usr --extra-version=0ubuntu0.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x556c987edf40] moov atom not found
core lecture 2.mp4: Invalid data found when processing input
```
The odd thing is, I have been able to run the code once before (the implied previous "lecture 1") - admittedly, "lecture 1" also failed to transcribe the first two times, only to unexpectedly transcribe the MP4 file's entire two hour duration on my next. And for the life of me, I cannot remember what was different when that third attempt worked... that string is long gone now, I foolishly deleted the cell when I tried to move on to transcribing "lecture 2."
After failing to get results at least five more times, (plus exiting the program, restarting the session, etc.), I think I'm in way over my head. Please tell me plainly where I have gone wrong, and if you have answers, I would greatly appreciate if you 1) gave me step-by-step, idiot-proof directions what to do, and 2) write any code I need **in its full form, exactly how it should be written in a cell** so I may paste it where needed.
Thank you for your time, and apologies in advance.

**Comment by phineas-pta** [comment]
your files is corrupted, you should test other files

**Reply by Tex2002ans** [reply]
> your files is corrupted, you should test other files

Yes, that may be likely. The final lines of the error says:

```
RuntimeError: Failed to load audio: ffmpeg version 4.2.7-0ubuntu0.1 [...]
[...]
The above exception was the direct cause of the following exception:
[...]
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x556c987edf40] moov atom not found
core lecture 2.mp4: Invalid data found when processing input
```

perplexed-paperclip may want to look up:

- ffmpeg's `moov atom not found`.
 
Maybe there's a way to try to "correct" your current audio file. (If you can go back to the source and redownload the MP4, that may be easier.)

Also, I see that:

- Your ffmpeg is **4.2.7**.

ffmpeg 6.0(!!!) is out now. If you update, perhaps some bugs were fixed since 4.2.7 which may help.

But definitely try correcting/redownloading that MP4 file first!
